dali/dfu/dfurun.cpp

```diff
@@ -540,23 +540,28 @@ class CDFUengine: public CInterface, implements IDFUengine
         return result;
     }
 
-    void ensureFilePermissions(const char * planeName, const char * fileName, SecAccessFlags perm, bool write)
+    void ensureFilePermissions(const char * planeName, const char * fileName, SecAccessFlags perm, IUserDescriptor *user, bool write)
     {
         if ((write && !HASWRITEPERMISSION(perm)) || (!write && !HASREADPERMISSION(perm)))
         {
+            StringBuffer context;
+            StringBuffer username("Null User");
+            if (user != nullptr) user->getUserName(username);
```

Comment:
getXXX methods that fill a StringBuffer usually append (not set), that looks to be the case for getUserName too:
```
    StringBuffer &getUserName(StringBuffer &buf)
    {
        return buf.append(username);
    }
```

Would also be better not to prepopulate with "Null User" and then reset.
I'd change to:
```
StringBuffer username;
if (user)
    user->getUserName(username);
else
    username.append("Null user");
```